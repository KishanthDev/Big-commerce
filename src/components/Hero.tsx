"use client";

import Image from "next/image";
import { Button } from "@/components/ui/button";

export default function HeroSection() {
  return (
    <section className="relative flex flex-col md:flex-row items-center justify-between min-h-screen px-8 py-16 lg:px-24 overflow-hidden">
      <div className="absolute inset-0 bg-gradient-to-br from-blue-500/40 via-transparent to-transparent pointer-events-none"></div>

      <div className="max-w-2xl text-center md:text-left z-10">
        <h1 className="text-5xl md:text-6xl font-bold leading-tight">
          It's your future.<br/> Shape it on
          <span className="text-blue-600"> your <br/>terms.</span>
        </h1>
        <p className="mt-6 text-xl text-black">
          Think big — and grow bigger — with our flexible, professional-grade ecommerce platform.
        </p>
        <div className="mt-8 flex flex-col md:flex-row space-y-4 md:space-y-0 md:space-x-6">
          <Button className="px-8 py-6 text-xl text-white bg-blue-600 rounded-xl shadow-lg">
            Explore Platform
          </Button>
          <Button variant="outline" className="px-8 py-6 text-xl border-primary text-primary rounded-xl">
            Get Started
          </Button>
        </div>
      </div>

      <div className="relative mt-12 md:mt-0">
        <Image src="/cube.png" alt="Cube" width={900} height={900} />
      </div>
    </section>
  );
}
