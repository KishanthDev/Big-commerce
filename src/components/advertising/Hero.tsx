import { ArrowRight } from "lucide-react";
import { Button } from "../ui/button";

export default function Hero() {
  return (
    <section className="bg-gradient-to-br from-blue-500/40 dark:from-blue-700/30 via-transparent to-transparent text-white p-6 md:p-16 lg:p-28">
      <div className="container mx-auto">
        <div className="flex flex-col md:flex-row items-center gap-10">
          <div className="flex-1 text-center md:text-left">
            <h1 className="text-3xl sm:text-4xl md:text-5xl text-primary font-bold mb-4">
              Promote Your Store to Thousands of Buyers
            </h1>
            <p className="text-base sm:text-lg text-primary opacity-90 mb-6 max-w-lg mx-auto md:mx-0">
              Get noticed with our affordable advertising options. Reach active shoppers looking for products like yours and increase your store's visibility on MarketHub.
            </p>
            <Button
              variant="outline"
              className="px-6 py-4 sm:px-8 sm:py-6 text-base sm:text-lg border-primary text-primary rounded-xl"
            >
              Advertise Today <ArrowRight className="w-4 h-4 ml-2" />
            </Button>
          </div>

          {/* Image/Visual Block */}
          <div className="flex-1 w-full h-60 sm:h-72 md:h-80 bg-gray-200 rounded-lg flex items-center justify-center text-gray-500 text-sm sm:text-base">
            <span>Advertising Campaign Image</span>
          </div>
        </div>
      </div>
    </section>
  );
}
