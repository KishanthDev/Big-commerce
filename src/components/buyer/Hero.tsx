import { ArrowRight } from 'lucide-react';

export default function Hero() {
  return (
    <section className="bg-blue-600 dark:bg-black text-white py-16">
      <div className="container mx-auto px-5">
        <div className="flex flex-col md:flex-row items-center gap-10">
          <div className="flex-1">
            <h1 className="text-4xl md:text-5xl font-bold mb-4">Discover Unique Products</h1>
            <p className="text-lg opacity-90 mb-6 max-w-lg">
              Shop directly from independent creators and boutique stores all in one place. Support local businesses while finding products you'll love.
            </p>
            <button className="bg-white dark:bg-gray-300 text-blue-600 px-5 py-2 rounded-md font-medium hover:bg-gray-100 transition inline-flex items-center">
              Get Started <ArrowRight className="w-4 h-4 ml-2" />
            </button>
          </div>
          <div className="flex-1 bg-gray-200 h-80 rounded-lg flex items-center justify-center text-gray-500">
            {/* Replace with actual image */}
            <span>Shopping Experience Image</span>
          </div>
        </div>
      </div>
    </section>
  );
}