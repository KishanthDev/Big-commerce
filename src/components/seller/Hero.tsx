import { ArrowRight } from "lucide-react";
import { Button } from "../ui/button";

export default function Hero() {
  return (
    <section className="bg-gradient-to-br from-blue-500/40 dark:from-blue-700/30 via-transparent to-transparent text-white p-6 md:p-16 lg:p-28">
      <div className="container mx-auto">
        <div className="flex flex-col md:flex-row items-center gap-8 md:gap-10">
          <div className="flex-1 text-center md:text-left">
            <h1 className="text-3xl sm:text-4xl md:text-5xl text-primary font-bold mb-4">
              Grow Your Business
            </h1>
            <p className="text-base sm:text-lg text-primary opacity-90 mb-6 max-w-md mx-auto md:mx-0">
              Join the platform that connects local sellers with nearby
              customers, all without a hassle
            </p>
            <div className="flex justify-center md:justify-start">
              <Button
                variant="outline"
                className="px-6 py-4 sm:px-8 sm:py-6 text-lg sm:text-xl border-primary text-primary rounded-xl"
              >
                Get Started <ArrowRight className="w-4 h-4 ml-2" />
              </Button>
            </div>
          </div>
          <div className="flex-1 bg-gray-200 h-60 sm:h-72 md:h-80 rounded-lg flex items-center justify-center text-gray-500 text-sm sm:text-base">
            <span>Selling Experience Image</span>
          </div>
        </div>
      </div>
    </section>
  );
}
