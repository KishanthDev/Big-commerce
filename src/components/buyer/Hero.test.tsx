import { render, screen } from "@testing-library/react";
import Hero from "./Hero";
import { ArrowRight } from "lucide-react";

jest.mock("lucide-react", () => ({
  ArrowRight: jest.fn(() => <svg data-testid="arrow-right" />),
}));

describe("Hero Component", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test("displays the main heading", () => {
    render(<Hero />);
    expect(
      screen.getByRole("heading", { name: /Discover Unique Products/i }),
    ).toBeInTheDocument();
  });

  test("shows the subheading text", () => {
    render(<Hero />);
    expect(
      screen.getByText(
        /Shop directly from independent creators and boutique stores/i,
      ),
    ).toBeInTheDocument();
  });

  test("renders the CTA button with correct text", () => {
    render(<Hero />);
    const button = screen.getByRole("button", { name: /Get Started/i });
    expect(button).toBeInTheDocument();
    expect(button).toHaveTextContent("Get Started");
  });

  test("includes ArrowRight icon in the button", () => {
    render(<Hero />);
    expect(ArrowRight).toHaveBeenCalled();
    expect(screen.getByTestId("arrow-right")).toBeInTheDocument();
  });

  test("renders the image placeholder", () => {
    render(<Hero />);
    expect(screen.getByText(/Shopping Experience Image/i)).toBeInTheDocument();
  });

  test("button has correct styling classes", () => {
    render(<Hero />);
    const button = screen.getByRole("button");
    expect(button).toHaveClass("border-primary");
    expect(button).toHaveClass("text-primary");
    expect(button).toHaveClass("rounded-xl");
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<Hero />);
    expect(asFragment()).toMatchSnapshot();
  });
});
