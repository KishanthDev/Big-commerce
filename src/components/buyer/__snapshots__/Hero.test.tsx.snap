// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Component matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="bg-gradient-to-br from-blue-500/40 dark:from-blue-700/30 via-transparent to-transparent text-white p-6 md:p-16 lg:p-28"
  >
    <div
      class="container mx-auto"
    >
      <div
        class="flex flex-col md:flex-row items-center gap-10"
      >
        <div
          class="flex-1 text-center md:text-left"
        >
          <h1
            class="text-3xl sm:text-4xl md:text-5xl text-primary font-bold mb-4"
          >
            Discover Unique Products
          </h1>
          <p
            class="text-base sm:text-lg text-primary opacity-90 mb-6 max-w-lg mx-auto md:mx-0"
          >
            Shop directly from independent creators and boutique stores all in one place. Support local businesses while finding products you'll love.
          </p>
          <button
            class="inline-flex items-center justify-center gap-2 whitespace-nowrap font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive border bg-background shadow-xs hover:bg-accent hover:text-accent-foreground dark:bg-input/30 dark:border-input dark:hover:bg-input/50 h-9 has-[>svg]:px-3 px-6 py-4 sm:px-8 sm:py-6 text-base sm:text-lg border-primary text-primary rounded-xl"
            data-slot="button"
          >
            Get Started 
            <svg
              data-testid="arrow-right"
            />
          </button>
        </div>
        <div
          class="flex-1 w-full h-60 sm:h-72 md:h-80 bg-gray-200 rounded-lg flex items-center justify-center text-gray-500 text-sm sm:text-base"
        >
          <span>
            Shopping Experience Image
          </span>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
